### PR TITLE
Add `module_name` config setting

### PR DESCRIPTION
**Pull request by gkoz**

### CODE REVIEW

File: src/codegen/objects.rs [hunk 1]
```diff
@@ -23,13 +23,17 @@ pub fn generate(env: &Env, root_path: &Path, mod_rs: &mut Vec<String>, traits: &
             }
         };
 
-        let path = root_path.join(file_name(&class_analysis.full_name));
+        let mod_name = obj.module_name.clone().unwrap_or_else(|| {
```
Comment (EPashkin) [inline src/codegen/objects.rs]: May be better update `nameutils::file_name` to accept `&class_analysis.full_name` as `full_name` and `obj.module_name` as `configured_name` ?

Comment (gkoz) [inline src/codegen/objects.rs]: `file_name` just doesn't seem that useful. It derives the module name (this should not be its concern and has caused code duplication in practice) and puts an extension on it (which is trivial).

Comment (gkoz) [inline src/codegen/objects.rs]: We could move the first part into a function perhaps.

**Comment by gkoz** [comment]
@homu r+

**Pull request merged**